openstack/networking/v2/extensions/bgpvpns/results.go

```diff
@@ -0,0 +1,517 @@
+package bgpvpns
+
+import (
+	"net/url"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/pagination"
+)
+
+const (
+	invalidMarker = "-1"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract is a function that accepts a result and extracts a BGP VPN resource.
+func (r commonResult) Extract() (*BGPVPN, error) {
+	var s BGPVPN
+	err := r.ExtractInto(&s)
+	return &s, err
+}
+
+func (r commonResult) ExtractInto(v any) error {
+	return r.Result.ExtractIntoStructPtr(v, "bgpvpn")
+}
+
+// BGPVPN represents an MPLS network with which Neutron routers and/or networks
+// may be associated
+type BGPVPN struct {
+	// The ID of the BGP VPN.
+	ID string `json:"id"`
+
+	// The user meaningful name of the BGP VPN.
+	Name string `json:"name"`
+
+	// Selection of the type of VPN and the technology behind it. Allowed
+	// values are l2 or l3.
+	Type string `json:"type"`
+
+	// Indicates whether this BGP VPN is shared across tenants.
+	Shared bool `json:"shared"`
+
+	// List of route distinguisher strings. If this parameter is specified,
+	// one of these RDs will be used to advertise VPN routes.
+	RouteDistinguishers []string `json:"route_distinguishers"`
+
+	// Route Targets that will be both imported and used for export.
+	RouteTargets []string `json:"route_targets"`
+
+	// Additional Route Targets that will be imported.
+	ImportTargets []string `json:"import_targets"`
+
+	// Additional Route Targets that will be used for export.
+	ExportTargets []string `json:"export_targets"`
+
+	// This read-only list of network IDs reflects the associations defined
+	// by Network association API resources.
+	Networks []string `json:"networks"`
+
+	// This read-only list of router IDs reflects the associations defined
+	// by Router association API resources.
+	Routers []string `json:"routers"`
+
+	// This read-only list of port IDs reflects the associations defined by
+	// Port association API resources (only present if the
+	// bgpvpn-routes-control API extension is enabled).
+	Ports []string `json:"ports"`
+
+	// The default BGP LOCAL_PREF of routes that will be advertised to the
+	// BGPVPN (unless overridden per-route).
+	LocalPref *int `json:"local_pref"`
```

Comment:
Why is it a pointer?

Reply:
Because we need to have an ability to set it to 0 on update.